File: 03-architecture/imooc-music-app-refactor/imooc_voice/src/main/java/com/imooc/imooc_voice/application/ImoocVoiceApplication.java
package com.imooc.imooc_voice.application;

import android.app.Application;

import com.alibaba.android.arouter.launcher.ARouter;
import com.imooc.lib_audio.app.AudioHelper;
import com.imooc.lib_share.share.ShareManager;
import com.imooc.lib_update.app.UpdateHelper;
import com.imooc.lib_video.app.VideoHelper;

public class ImoocVoiceApplication extends Application {

  private static ImoocVoiceApplication mApplication = null;

  @Override public void onCreate() {
    super.onCreate();
    mApplication = this;
    //视频SDK初始化
    VideoHelper.init(this);
    //音频SDK初始化
    AudioHelper.init(this);
    //分享SDK初始化
    ShareManager.initSDK(this);
    //更新组件下载
    UpdateHelper.init(this);
    //ARouter初始化
    ARouter.init(this);
  }

  public static ImoocVoiceApplication getInstance() {
    return mApplication;
  }
}

File: 03-architecture/imooc-music-app/imooc_voice/src/main/java/com/imooc/imooc_voice/application/ImoocVoiceApplication.kt
package com.imooc.imooc_voice.application

import android.app.Application
import com.alibaba.android.arouter.launcher.ARouter
import com.imooc.lib_audio.app.AudioHelper
import com.imooc.lib_share.share.ShareManager
import com.imooc.lib_update.app.UpdateHelper
import com.imooc.lib_video.app.VideoHelper

class ImoocVoiceApplication : Application() {

    //伴身对象实现单例
    companion object {
        lateinit var mInstance: ImoocVoiceApplication
    }

    override fun onCreate() {
        super.onCreate()
        mInstance = this
        init()
    }

    private fun init() {
        //视频SDK初始化
        VideoHelper.init(mInstance)
        //音频SDK初始化
        AudioHelper.init(mInstance)
        //分享SDK初始化
        ShareManager.initSDK(mInstance)
        //更新组件下载
        UpdateHelper.init(mInstance)
        //ARouter初始化
        ARouter.init(mInstance)
    }
}